### PR TITLE
update go-critic to v0.5.6

### PR DESCRIPTION
**Pull request by pierrre**
Fix https://github.com/golangci/golangci-lint/issues/1923

### CODE REVIEW

File: pkg/golinters/govet_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 
 func TestGovet(t *testing.T) {
 	// Checking that every default analyzer is in "all analyzers" list.
-	checkList := append(defaultAnalyzers,
+	checkList := append(defaultAnalyzers, //nolint:gocritic // Build a temporary list of analyzers.
```
Comment (quasilyte) [inline pkg/golinters/govet_test.go:21]: Depending on the code style used in this project, there are multiple solutions.
Given that the original code looks like this:
```go
	checkList := append(defaultAnalyzers,
		shadow.Analyzer, // special case, used in analyzersFromConfig
	)
```
We can do something from the options below.
```go
checkList := defaultAnalyzers
checkList = append(checkList, shadow.Analyzer)
```
```go
var checkList []*analysis.Analyzer
checkList = append(checkList, defaultAnalyzers...)
checkList = append(checkList, shadow.Analyzer)
```
I think there are more alternatives available, but these should suffice.

File: go.mod [hunk 2]
```diff
@@ -18,7 +18,8 @@ require (
 	github.com/esimonov/ifshort v1.0.2
 	github.com/fatih/color v1.10.0
 	github.com/fzipp/gocyclo v0.3.1
-	github.com/go-critic/go-critic v0.5.5
+	github.com/go-critic/go-critic v0.5.6
+	github.com/go-toolsmith/pkgload v1.0.1 // indirect
```
Comment (quasilyte) [inline go.mod:22]: I wonder if `go mod tidy` would remove these indirect deps.
I don't think they're required.

**Comment by ldez** [comment]
`go-ruleguard` uses `strconv.ParseComplex`, this function was introtuced in go1.15.

Our CI, run with go1.14, go1.15, go1.16, then it's an issue with go1.14.

```console
$ go build -o golangci-lint ./cmd/golangci-lint
go: downloading github.com/go-critic/go-critic v0.5.6
go: downloading github.com/quasilyte/go-ruleguard v0.3.4
# github.com/quasilyte/go-ruleguard/internal/gogrep
Error: ../../../go/pkg/mod/github.com/quasilyte/go-ruleguard@v0.3.4/internal/gogrep/match.go:677:13: undefined: strconv.ParseComplex
note: module requires Go 1.15
make: *** [Makefile:61: golangci-lint] Error 2
Error: Process completed with exit code 2.
```

File: go.mod (hunk 2 above)
Comment (pierrre) [reply go.mod]: I don't know. I think I ran `go mod tidy`.

**Comment by pierrre** [comment]
What should we do ?

I look at the previous code, and the older version of `go-ruleguard` was not using `strconv.ParseComplex()`. (it was not parsing complex at all)

Is there a problem to update the minimum Go version required for golangci-lint ?
Is it a problem for people who run `golangci-lint` as a command line ?

**Comment by ldez** [comment]
I need time to think about it.

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**